### DISCUSSION TITLE
API - POST ​/pdustats error

### DISCUSSION THREAD

**Discussion by MathisICDC** [opendcim/openDCIM#1547] [Q&A]
Hello,
I'm encontring an error "401 Access Denied" when i run my python script. 
I'm only getting this error on this script, even though I already have 5 of them running every day. I use the same POST request and yet I still get this error. I'm also using the same APIKey account for all the scripts, as I'm using SAML authentication.
When I do it via swagger, there's no problem but via Postman or script impossible.
Is anyone experiencing the same problem with this API POST ​/pdustats request?

Here's the code I'm trying to run :
![image](https://github.com/user-attachments/assets/94e023dc-d070-48b2-aba5-8da9a60de2c3)

Below is the result of my test script when I run the previous code:
```
C:\Users\X\Projet\Projet DCIM\Script\Python>python test.py
{"PDUID": 3757,"Wattage": 1200,"LastRead": "2024-07-12 17:36:07"}
https://xxxxxxxx/api/v1/pdustats
{"error":true,"errorcode":401,"message":"Access Denied"}
```
Thank you for your help !

**Comment by samilliken** [comment]
When using swagger are you using a private window and entering your key?

Scott

**Reply by MathisICDC** [reply]
Yes, I use a private window and put in my login and password. 
I found the error. I added rights to the API account and it worked. I didn't know that the rights of an account also came into play when using the API service.

Mathis
